duckdb: 0.8.1 -> 0.9.0

purcell opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

purcell committed:
duckdb: 0.8.1 -> 0.9.0

purcell commented:
CC'ing maintainers: @costrouc @cpcloud

I haven't gone through the release to see if there are extra features that we should enable: figured it was useful just to bump the version initially and check everything still works.

I get a failure in `nix-review` locally because the duckdb Common Lisp package (`cl-duckdb`, maintainer @ak-coram) doesn't build on my M2 Mac — I think the fault lies in `cffi-libffi` though, not the binding itself:
```
* LOAD-SYSTEMS
* ; compiling file "/nix/store/vdli1g6yiwp0xcaiw9qs44xnphnrgnby-source/libffi/libffi.lisp" (written 01 JAN 1970 12:00:01 AM):

; wrote /private/tmp/nix-build-cffi-libffi-20230618-git.drv-0/source/libffi/libffi-tmpGHU3ALSV.fasl
; compilation finished in 0:00:00.003
; pkg-config libffi --cflags
; ERROR: Couldn't execute "pkg-config": No such file or directory


; Attempting to continue anyway.
; clang -o /private/tmp/nix-build-cffi-libffi-20230618-git.drv-0/source/libffi/libffi-types__grovel-tmpAAURSO1.o -c -g -Wall -Wundef -Wsign-compare -Wpointer-arith -O3 -g -Wall -fdollars-in-identifiers -fno-omit-frame-pointer -I/opt/local/include/ -fPIC -I/nix/store/saf0bvf2gn4m8sqpzsw17vd0fhlmh7fs-cffi-grovel-20230618-git/ /private/tmp/nix-build-cffi-libffi-20230618-git.drv-0/source/libffi/libffi-types__grovel.c
/private/tmp/nix-build-cffi-libffi-20230618-git.drv-0/source/libffi/libffi-types__grovel.c:6:10: fatal error: 'ffi/ffi.h' file not found
#include <ffi/ffi.h>
         ^~~~~~~~~~~
1 error generated.
;
; compilation unit aborted
;   caught 1 fatal ERROR condition
BUILD FAILED: Subprocess #<UIOP/LAUNCH-PROGRAM:PROCESS-INFO {7006E27143}>
 with command ("clang" "-o"
               "/private/tmp/nix-build-cffi-libffi-20230618-git.drv-0/source/libffi/libffi-types__grovel-tmpAAURSO1.o"
               "-c" "-g" "-Wall" "-Wundef" "-Wsign-compare" "-Wpointer-arith"
               "-O3" "-g" "-Wall" "-fdollars-in-identifiers"
               "-fno-omit-frame-pointer" "-I/opt/local/include/" "-fPIC"
               "-I/nix/store/saf0bvf2gn4m8sqpzsw17vd0fhlmh7fs-cffi-grovel-20230618-git/"
               "/private/tmp/nix-build-cffi-libffi-20230618-git.drv-0/source/libffi/libffi-types__grovel.c")
 exited with error code 1
```

purcell commented:
Also python3.11-duckdb-0.9.0 failing locally:
```
@nix { "action": "setPhase", "phase": "pythonCatchConflictsPhase" }
pythonCatchConflictsPhase
/nix/store/q2zpnf017gccs9w9mb0ikdqhkf0k5v82-catch_conflicts.py:1: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
  import pkg_resources
@nix { "action": "setPhase", "phase": "pythonRemoveBinBytecodePhase" }
pythonRemoveBinBytecodePhase
@nix { "action": "setPhase", "phase": "pythonImportsCheckPhase" }
pythonImportsCheckPhase
Executing pythonImportsCheckPhase
Check whether the following modules can be imported: duckdb
@nix { "action": "setPhase", "phase": "pytestCheckPhase" }
pytestCheckPhase
Executing pytestCheckPhase
ImportError while loading conftest '/private/tmp/nix-build-python3.11-duckdb-0.9.0.drv-0/source/tools/pythonpkg/tests/conftest.py'.
tests/conftest.py:6: in <module>
    import duckdb
duckdb/__init__.py:4: in <module>
    import duckdb.functional as functional
duckdb/functional/__init__.py:1: in <module>
    from duckdb.duckdb.functional import (
E   ModuleNotFoundError: No module named 'duckdb.duckdb'
/nix/store/x3s84jwqal1gqhc5w7qx291fm5vqqjxh-stdenv-darwin/setup: line 1605: pop_var_context: head of shell_variables not a function context
```

ak-coram commented:
@purcell I can confirm the CL bindings work just fine with the 0.9.0 DuckDB release in [various configurations](https://github.com/ak-coram/cl-duckdb/actions/runs/6315686113). The bindings had substantial upgrades in the last six months, so they might be worth updating along with DuckDB itself. Multiple CL dependencies have been dropped, but there are also some new ones not yet packaged for NixOS.

As for the build error: `Couldn't execute "pkg-config": No such file or directory` is rather common and doesn't usually cause issues, but perhaps `pkg-config` might be able to provide the right flags for the compiler so that the other error goes away. I'm too unfamiliar with Nix to reproduce and test this quickly.

Not sure whether only SBCL is supported, but the new ECL release 23.9.9 has some known [GC issues](https://gitlab.com/catap/ecl/-/commit/e8c9355d31e15d15900416e9925db56c54c9f265)  on macOS causing sporadic failures even during compilation. You might want to keep that in mind.

Let me know if I can help out in any way!

cpcloud commented:
@purcell I can help with fixing the Python bindings. Should I make a PR to your fork?

cpcloud commented:
FWIW I am able to build and run duckdb 0.9 on `x86_64-linux` without issue

cpcloud commented:
@purcell Instead of submitting a PR to your fork, since the changes are very small, here's a patch:

```patch
diff --git a/pkgs/development/python-modules/duckdb/default.nix b/pkgs/development/python-modules/duckdb/default.nix
index f582ae9c3fdf..8989bd5d46c1 100644
--- a/pkgs/development/python-modules/duckdb/default.nix
+++ b/pkgs/development/python-modules/duckdb/default.nix
@@ -1,6 +1,7 @@
 { lib
 , buildPythonPackage
 , duckdb
+, fsspec
 , google-cloud-storage
 , numpy
 , pandas
@@ -44,6 +45,7 @@ buildPythonPackage rec {
     google-cloud-storage
     psutil
     pytestCheckHook
+    fsspec
   ];

   disabledTests = [
@@ -55,6 +57,10 @@ buildPythonPackage rec {
     export HOME="$(mktemp -d)"
   '';

+  setupPyBuildFlags = [
+    "--inplace"
+  ];
+
   pythonImportsCheck = [
     "duckdb"
   ];
```

purcell commented:
I just incorporated your patch and re-pushed, thanks. But the common lisp side will remain broken, and it looks like to resolve that we might need to update the quicklisp distro. I started updating a couple of the deps for cl-duckdb by hand and it didn't feel like the way to go.

purcell commented:
@ak-coram Your comment was super helpful, thanks! I think we're just dealing with sbcl support here, not ecl. There's a frozen set of quicklisp packages that get periodically updated, unsure who on the nixpkgs side would make that happen.

Commit:
update duckdb and python duckdb to v0.9.0

Commit:
make duckdb-engine build with duckdb 0.9.0. disable autoload and autoinstall as they seem buggy -- even if extension is built in they can download and load an extension automatically.

Commit:
expand tabs

purcell committed:
cl-duckdb: fix build for duckdb 0.9

purcell commented:
I pushed some changes to try to get `cl-duckdb` building, which included adding @ak-coram's `frugal-uuid` and `trivial-clock` packages. I didn't manage to get the latter building, strangely: `nix-build -A sbclPackages.trivial-clock` reports the following:
```
this derivation will be built:
  /nix/store/fw15hlj77fa88wl5wqgjfgi03x8982p5-trivial-clock-trunk.drv
building '/nix/store/fw15hlj77fa88wl5wqgjfgi03x8982p5-trivial-clock-trunk.drv'...
unpacking sources
unpacking source archive /nix/store/17c0piki2fd932r63z8mlar2i1p27hbm-source
source root is source
patching sources
updateAutotoolsGnuConfigScriptsPhase
configuring
no configure script, doing nothing
building
This is SBCL 2.3.8.nixos, an implementation of ANSI Common Lisp.
More information about SBCL is available at <http://www.sbcl.org/>.

SBCL is free software, provided as is, with absolutely no warranty.
It is mostly in the public domain; some portions are provided under
BSD-style licenses.  See the CREDITS and COPYING files in the
distribution for more information.
* #<PACKAGE "NIX-CL-USER">
* #<PACKAGE "NIX-CL-USER">
* T
* LOAD-SYSTEMS
* BUILD FAILED: Component "trivial-clock" not found
error: builder for '/nix/store/fw15hlj77fa88wl5wqgjfgi03x8982p5-trivial-clock-trunk.drv' failed with exit code 1;
       last 10 log lines:
       >
       > SBCL is free software, provided as is, with absolutely no warranty.
       > It is mostly in the public domain; some portions are provided under
       > BSD-style licenses.  See the CREDITS and COPYING files in the
       > distribution for more information.
       > * #<PACKAGE "NIX-CL-USER">
       > * #<PACKAGE "NIX-CL-USER">
       > * T
       > * LOAD-SYSTEMS
       > * BUILD FAILED: Component "trivial-clock" not found
```
Any guesses? Obviously with Nix we're manually adding in the depended-upon libraries, but I can't see that I've messed anything up.

purcell commented:
Ignore me, I messed something stupid up there. I've moved on to the next error.

purcell committed:
cffi-libffi: use ffi.h patch for darwin

This patch was in lisp-modules-new-obsolete, but not applied
here. Without it, cffi-libffi fails to build on darwin.

purcell commented:
I got `cl-duckdb` building now, but it would be good if someone familiar with the CL stuff in nixpkgs could review what I've done, perhaps @Uthar?

Commit:
use the full hash and shallow git clone

Commit:
skip failing duckdb tests.

robert-manchester commented:
see also PR #258203. The build changed with respect to extensions so this change in its current form is no longer building the built in extensions (or enabling the autoloaded ones). see #258203 for the modified cmake flags. that PR adds the hash to the version, which makes it slightly nicer when filing bugs, but most of the other changes in that PR are pretty minor.

purcell committed:
duckdb: Incorporate changes from #257452

purcell commented:
Thanks, I'm manually adding in your changes to this PR, hope that's okay with you.

purcell commented:
> might want to disable the autoloading in the python package (it is enabled by default there)

Would that be a patch to the python source?

robert-manchester commented:
> > might want to disable the autoloading in the python package (it is enabled by default there)
> 
> Would that be a patch to the python source?

yes. the patch removes the line that adds the two defines.

robert-manchester commented:
> Right, thanks. The issue I'm having locally with these changes is that I can't get `duckdb` to skip the `test_remote_optimizer` test. :(

this one?
"Test using a remote optimizer pass in case thats important to someone"
can't just add that string in the exclude list? i don't see any .test files with that name or functions in the python dir.

purcell commented:
Thanks, yes, that works, but then another pops up which I'd be hesitant to just exclude:
```
Test closing database during long running query
-------------------------------------------------------------------------------
/tmp/nix-build-duckdb-0.9.0.drv-3/source/test/api/test_api.cpp:68
```

robert-manchester commented:
> test/api/test_api.cpp

that was skipped before too, i should've left it disabled. -- looking at it is a racy test anyway. not sure why it is failing on your system but looks like it is possible that the query can finish before the test thread interrupts it and thus will fail the test.

robert-manchester commented on pkgs/development/python-modules/duckdb/default.nix:
```diff
@@ -53,8 +62,14 @@ buildPythonPackage rec {
 
   preCheck = ''
     export HOME="$(mktemp -d)"
+    # duckdb directory prevents loading the python module
+    mv duckdb duckdb.pk
```
can probably get rid of these two lines. the `--inplace` added to `setupPyBuildFlags` may be the proper fix for this.

purcell replied:
Done, thanks

purcell commented:
> that was skipped before too, i should've left it disabled. -- looking at it is a racy test anyway

Adding that exclusion back in.

purcell commented:
Okay, I feel pretty good about the state of things now — this PR can build `duckdb`, `python310Packages.duckdb`, `python310Packages.duckdb-engine`, `python310Packages.ibis-framework`, `sbclPackages.duckdb` on my `aarch64-darwin` machine. Thanks for the pointers, @robert-manchester!

purcell commented:
Running nix-review locally is a pain, because so many of the dependent packages (e.g. those lisp packages depending on cffi-libffi) just don't build on my platform anyway. But perhaps someone on Linux would be able to fire up a run. (Does ofborg support this?)

danderson commented:
Tried to do a nix-review PR run on my linux box, and it failed with a syntax error:

<details>
<summary>syntax error, unexpected DOLLAR_CURLY, expecting ';', at /home/dave/.cache/nixpkgs-review/pr-257452-2/nixpkgs/pkgs/development/compilers/cudatoolkit/redist/extension.nix:92:65</summary>
```
error: while evaluating anonymous function at /home/dave/.cache/nixpkgs-review/pr-257452-2/nixpkgs/lib/customisation.nix:82:32, called from /home/dave/.cache/nixpkgs-review/pr-257452-2/nixpkgs/pkgs/top-level/all-packages.nix:15554:20:
while evaluating 'makeOverridable' at /home/dave/.cache/nixpkgs-review/pr-257452-2/nixpkgs/lib/customisation.nix:72:24, called from /home/dave/.cache/nixpkgs-review/pr-257452-2/nixpkgs/lib/customisation.nix:82:41:
while evaluating anonymous function at /home/dave/.cache/nixpkgs-review/pr-257452-2/nixpkgs/pkgs/applications/science/misc/colmap/default.nix:1:1, called from /home/dave/.cache/nixpkgs-review/pr-257452-2/nixpkgs/lib/customisation.nix:74:16:
while evaluating 'recurseIntoAttrs' at /home/dave/.cache/nixpkgs-review/pr-257452-2/nixpkgs/lib/attrsets.nix:1018:5, called from /home/dave/.cache/nixpkgs-review/pr-257452-2/nixpkgs/pkgs/top-level/all-packages.nix:7276:18:
while evaluating 'callPackageWith' at /home/dave/.cache/nixpkgs-review/pr-257452-2/nixpkgs/lib/customisation.nix:122:35, called from /home/dave/.cache/nixpkgs-review/pr-257452-2/nixpkgs/pkgs/top-level/all-packages.nix:7266:23:
while evaluating 'makeOverridable' at /home/dave/.cache/nixpkgs-review/pr-257452-2/nixpkgs/lib/customisation.nix:72:24, called from /home/dave/.cache/nixpkgs-review/pr-257452-2/nixpkgs/lib/customisation.nix:173:34:
while evaluating anonymous function at /home/dave/.cache/nixpkgs-review/pr-257452-2/nixpkgs/pkgs/top-level/cuda-packages.nix:1:1, called from /home/dave/.cache/nixpkgs-review/pr-257452-2/nixpkgs/lib/customisation.nix:74:16:
while evaluating 'overrideScope' at /home/dave/.cache/nixpkgs-review/pr-257452-2/nixpkgs/lib/customisation.nix:272:27, called from /home/dave/.cache/nixpkgs-review/pr-257452-2/nixpkgs/pkgs/top-level/cuda-packages.nix:75:5:
while evaluating 'makeScope' at /home/dave/.cache/nixpkgs-review/pr-257452-2/nixpkgs/lib/customisation.nix:268:25, called from /home/dave/.cache/nixpkgs-review/pr-257452-2/nixpkgs/lib/customisation.nix:272:30:
while evaluating 'extends' at /home/dave/.cache/nixpkgs-review/pr-257452-2/nixpkgs/lib/fixed-points.nix:91:24, called from /home/dave/.cache/nixpkgs-review/pr-257452-2/nixpkgs/lib/customisation.nix:269:16:
while evaluating 'composeExtensions' at /home/dave/.cache/nixpkgs-review/pr-257452-2/nixpkgs/lib/fixed-points.nix:99:18, called from /home/dave/.cache/nixpkgs-review/pr-257452-2/nixpkgs/lib/fixed-points.nix:91:67:
while evaluating 'composeExtensions' at /home/dave/.cache/nixpkgs-review/pr-257452-2/nixpkgs/lib/fixed-points.nix:99:18, called from /home/dave/.cache/nixpkgs-review/pr-257452-2/nixpkgs/lib/fixed-points.nix:102:22:
while evaluating 'composeExtensions' at /home/dave/.cache/nixpkgs-review/pr-257452-2/nixpkgs/lib/fixed-points.nix:99:18, called from /home/dave/.cache/nixpkgs-review/pr-257452-2/nixpkgs/lib/fixed-points.nix:102:22:
syntax error, unexpected DOLLAR_CURLY, expecting ';', at /home/dave/.cache/nixpkgs-review/pr-257452-2/nixpkgs/pkgs/development/compilers/cudatoolkit/redist/extension.nix:92:65
$ git worktree prune
Traceback (most recent call last):
  File "/nix/store/y65klixc4d0pbab8mw5caw280w72620l-nixpkgs-review-2.4.2/bin/.nixpkgs-review-wrapped", line 9, in <module>
    sys.exit(main())
  File "/nix/store/y65klixc4d0pbab8mw5caw280w72620l-nixpkgs-review-2.4.2/lib/python3.8/site-packages/nixpkgs_review/__init__.py", line 10, in main
    cli.main(command, args)
  File "/nix/store/y65klixc4d0pbab8mw5caw280w72620l-nixpkgs-review-2.4.2/lib/python3.8/site-packages/nixpkgs_review/cli/__init__.py", line 228, in main
    args.func(args)
  File "/nix/store/y65klixc4d0pbab8mw5caw280w72620l-nixpkgs-review-2.4.2/lib/python3.8/site-packages/nixpkgs_review/cli/pr.py", line 60, in pr_command
    contexts.append((pr, review.build_pr(pr)))
  File "/nix/store/y65klixc4d0pbab8mw5caw280w72620l-nixpkgs-review-2.4.2/lib/python3.8/site-packages/nixpkgs_review/review.py", line 185, in build_pr
    return self.build_commit(base_rev, pr_rev)
  File "/nix/store/y65klixc4d0pbab8mw5caw280w72620l-nixpkgs-review-2.4.2/lib/python3.8/site-packages/nixpkgs_review/review.py", line 132, in build_commit
    base_packages = list_packages(str(self.worktree_dir()))
  File "/nix/store/y65klixc4d0pbab8mw5caw280w72620l-nixpkgs-review-2.4.2/lib/python3.8/site-packages/nixpkgs_review/review.py", line 287, in list_packages
    return parse_packages_xml(nix_env.stdout)
  File "/nix/store/y65klixc4d0pbab8mw5caw280w72620l-nixpkgs-review-2.4.2/lib/python3.8/site-packages/nixpkgs_review/review.py", line 228, in parse_packages_xml
    for (event, elem) in context:
  File "/nix/store/qy5z9gcld7dljm4i5hj3z8a9l6p37y81-python3-3.8.8/lib/python3.8/xml/etree/ElementTree.py", line 1233, in iterator
    root = pullparser._close_and_return_root()
  File "/nix/store/qy5z9gcld7dljm4i5hj3z8a9l6p37y81-python3-3.8.8/lib/python3.8/xml/etree/ElementTree.py", line 1280, in _close_and_return_root
    root = self._parser.close()
xml.etree.ElementTree.ParseError: no element found: line 1, column 0
```
</details>

Looks unrelated to this PR, so either something's broken with my system, or maybe the PR needs a pull --rebase onto master to pick up an unrelated fix?

ak-coram commented:
> But perhaps someone on Linux would be able to fire up a run.

@purcell I can take it for a spin on Linux later today, I've tried nix-review yesterday and even the original old version of cl-duckdb was building fine so I think the cffi-libffi error was probably macOS specific (the [other PR](https://github.com/NixOS/nixpkgs/pull/258203#issuecomment-1742066971) couldn't reproduce it either). Perhaps the packages didn't build on macOS even before the DuckDB update.

I have some automated tests for the bindings on AArch64 via GH actions & emulation, so I don't think it's the architecture (these are admittedly only Linux / ECL). The CL bindings even work on my cheap Android phone via both ECL and SBCL.

danderson commented:
Result of `nixpkgs-review pr 257452` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>20 packages failed to build:</summary>
  <ul>
    <li>duckdb</li>
    <li>mycli</li>
    <li>mycli.dist</li>
    <li>python310Packages.duckdb</li>
    <li>python310Packages.duckdb-engine</li>
    <li>python310Packages.duckdb-engine.dist</li>
    <li>python310Packages.duckdb.dist</li>
    <li>python310Packages.ibis-framework</li>
    <li>python310Packages.ibis-framework.dist</li>
    <li>python310Packages.sqlglot</li>
    <li>python310Packages.sqlglot.dist</li>
    <li>python311Packages.duckdb</li>
    <li>python311Packages.duckdb-engine</li>
    <li>python311Packages.duckdb-engine.dist</li>
    <li>python311Packages.duckdb.dist</li>
    <li>python311Packages.ibis-framework</li>
    <li>python311Packages.ibis-framework.dist</li>
    <li>python311Packages.sqlglot</li>
    <li>python311Packages.sqlglot.dist</li>
    <li>sbclPackages.duckdb</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>sbclPackages.frugal-uuid</li>
    <li>sbclPackages.trivial-clock</li>
  </ul>
</details>

danderson commented:
nixpkgs-review is giving me a different source hash than you on the checkout of the duckdb repository. I fear this is the longstanding thing where deepclone=true doesn't produce stable output :/ Unless there are known fixups for that, might have to go back to the previous version's strategy of patching the build process and using a shallow clone.

danderson commented:
```
error: hash mismatch in fixed-output derivation '/nix/store/y0dj5h0hfxypv2bp4n76m87741f40rc6-source.drv':
         specified: sha256-EKvDH7RwOC4Gu/lturrfnGpzXnJ9azIwAFeuVoa6L/Y=
            got:    sha256-NFkeeTpsxazQOstKUUu0b27hXbnq3U5g/+24BIMqtJY=
error: 1 dependencies of derivation '/nix/store/756bwwxy133wki95lhw1h10cvslw9wi9-duckdb-0.9.0.drv' failed to build
```

purcell commented:
> so I think the cffi-libffi error was probably macOS specific (the https://github.com/NixOS/nixpkgs/pull/258203#issuecomment-1742066971 couldn't reproduce it either). Perhaps the packages didn't build on macOS even before the DuckDB update.

@ak-coram That was my assumption too tbh.

> so either something's broken with my system, or maybe the PR needs a pull --rebase onto master to pick up an unrelated fix?

@danderson That is odd! Happy to rebase (I did so earlier today), but I don't see the same issue.

ak-coram commented:
@danderson I think you're right about the non-stable hash, I get the same error but with another hash result compared to yours:

```
error: hash mismatch in fixed-output derivation '/nix/store/y0dj5h0hfxypv2bp4n76m87741f40rc6-source.drv':
         specified: sha256-EKvDH7RwOC4Gu/lturrfnGpzXnJ9azIwAFeuVoa6L/Y=
         got:       sha256-Zptx8tNxEU3wCTmhkHW5bVs+lwf0YvWBJl1XOCkvMmg=
error: 1 dependencies of derivation '/nix/store/756bwwxy133wki95lhw1h10cvslw9wi9-duckdb-0.9.0.drv' failed to build
```

danderson commented:
@purcell the parse error thing was PEBKAC from me, I was using an old nixpkgs-review that couldn't parse modern nix source. The hash mismatch is real though :(

robert-manchester commented:
bummer. I had to add `test/sql/copy/csv/code_cov/csv_disk_reload.test` back into the exclude list. perhaps i was too optimistic that some of these tests were more stable than from 0.8.1. after that change all succeed for me.

seems like the merges from my PR need to get reverted for pkgs/development/python-modules/duckdb/default.nix and pkgs/development/libraries/duckdb/default.nix. only really need the cmake flags for the library, and then remove the autoinstall/autoload flags and add `  BUILD_HTTPFS = 1;` so the httpfs extension is built in for python.

Commit:
Merge branch 'duckdb_0.9.0_full_hash' into duckdb_0.9.0

Commit:
use --inplace for python build

Commit:
restore the original set of unittest excludes and add "Test using a remote optimizer pass in case thats important to someone"

Commit:
restore original version strategy with no git hash.

Commit:
httpfs extension was not bundled. just bundle all of the in tree extensions

Commit:
exclude additional httpfs tests which access network

Commit:
disable one more httpfs test test/sql/json/table/read_json_objects.test

Commit:
remove tabs

Commit:
Merge branch 'duckdb_0.9.0' into duckdb-0.9_nohash

robert-manchester commented:
FWIW i updated PR #258203 to restore the original unit tests excludes, eliminated deepClone to revert to not providing the commit hash to the build. I realized that the httpfs extension was not being built so modified the cmakeFlags so it is -- and removed additional unit tests that were using it to access the network.
nixpkgs-review passed. I made an attempt to create a diff to this PR -- merge ok, push to github looked wrong so killed it.

only hacky thing in that PR is that the library/duckdb/default.nix patch contains the patch used in the python module as i wasn't sure how to append the patches list when `patches` is provided by `inherit (duckdb) patches`.

Commit:
fix bad merge

Commit:
fix bad merge

Commit:
fix bad merge

purcell commented:
Nice. It might be a few days before I can look further at this but if you find yourself able to send a PR to this branch in my fork I'll happily merge and push it.

purcell committed:
Merge pull request #1 from robert-manchester/duckdb-0.9_nohash

duckdb 0.9.0 without git deepcopy and hash version.

purcell commented:
Changes from @robert-manchester now merged in (thanks!), this should have resolved the above issues.

ak-coram commented:
nix-review now successfully builds all 22 packages on Linux.

danderson commented:
Result of `nixpkgs-review pr 257452` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>22 packages built:</summary>
  <ul>
    <li>duckdb</li>
    <li>mycli</li>
    <li>mycli.dist</li>
    <li>python310Packages.duckdb</li>
    <li>python310Packages.duckdb-engine</li>
    <li>python310Packages.duckdb-engine.dist</li>
    <li>python310Packages.duckdb.dist</li>
    <li>python310Packages.ibis-framework</li>
    <li>python310Packages.ibis-framework.dist</li>
    <li>python310Packages.sqlglot</li>
    <li>python310Packages.sqlglot.dist</li>
    <li>python311Packages.duckdb</li>
    <li>python311Packages.duckdb-engine</li>
    <li>python311Packages.duckdb-engine.dist</li>
    <li>python311Packages.duckdb.dist</li>
    <li>python311Packages.ibis-framework</li>
    <li>python311Packages.ibis-framework.dist</li>
    <li>python311Packages.sqlglot</li>
    <li>python311Packages.sqlglot.dist</li>
    <li>sbclPackages.duckdb</li>
    <li>sbclPackages.frugal-uuid</li>
    <li>sbclPackages.trivial-clock</li>
  </ul>
</details>

purcell commented:
Yay, thanks all!
